spec/puppet-server-lib/puppet/jvm/http_client_spec.rb

```diff
@@ -5,11 +5,17 @@
 
 describe 'Puppet::Server::HttpClient' do
   let :client do
-    Puppet::Server::HttpClient.new("localhost", 0, {})
+    Puppet::Server::HttpClient.new
   end
 
+  let(:url) { 'http://localhost:0/' }
+
   context "when settings are initialized with specific values" do
     before :all do
+      Puppet[:hostcert] = "spec/fixtures/localhost-cert.pem"
+      Puppet[:hostprivkey] = "spec/fixtures/localhost-privkey.pem"
+      Puppet[:localcacert] = "spec/fixtures/ca-cert.pem"
```

Comment:
There was an order-dependent spec failure, where the [`config_spec`](https://github.com/puppetlabs/puppetserver/blob/6.x/spec/puppet-server-lib/puppet/jvm/config_spec.rb) sets these up and causes the `ssl_context` to be generated, which was then [memoized](https://github.com/puppetlabs/puppetserver/blob/6.x/src/ruby/puppetserver-lib/puppet/server/config.rb#L50-L56) and returned if this file ran after that one. But then this file couldn't be run in isolation.

Reply:
This fixes it.